[FE] IssueDetailHeader 컴포넌트 구현

Kakamotobi opened the pull request:
## Issues
- #79 

## What is this PR? 👓
- IssueDetailHeader 컴포넌트 구현.

## Key changes 🔑
- IssueDetailHeader에서 issue 제목, 열림/닫힘 수정 기능 추가.
- `TextInput` 컴포넌트에서 input으로 value 전달.
- Null check
  - 기존: `issueDetails && <IssueDetailHeader />`
  - 수정: `IssueDetailHeader` 내부에서 `issueDetails`가 null 일시에 초기 값 설정.

## To reviewers 👋
- `api/index.ts`에 `putIssueTitle` 및 `putIssueIsOpen` 함수의 `body`의 type이 너무 작아서 `api/type.ts`에 선언하지 않고 함수 인자에 바로 표시했습니다.

Kakamotobi committed:
#79 feat: Issue 제목, 열림/닫힘 API 추가

Kakamotobi committed:
#79 feat: Issue 제목, 열림/닫힘 수정 mock 추가

Kakamotobi committed:
#79 fix: TextInput의 input에 value 전달

Kakamotobi committed:
#79 feat: IssueDetailHeader 컴포넌트 구현

- 부모인 `IssueDetailPage`로부터 `issueDetails` 객체 및 수정 함수들을 props로 전달 받음.

Kakamotobi committed:
#79 refactor: issueDetails에 대한 null check을 IssueDetailHeader로 이동

- 기존: `issueDetails && <IssueDetailHeader />`
- 수정: `IssueDetailHeader` 내부에서 `issueDetails`가 null 일시에 초기 값 설정.

Kakamotobi committed:
#79 design: Label width 설정

youzysu commented on fe/src/components/Issues/IssueDetail/IssueDetailHeader.tsx:
```diff
@@ -0,0 +1,214 @@
+    isOpen: true,
+  };
+
+  useEffect(() => {
+    setNewTitle(title);
+  }, [title]);
```
issueDetails fetch 완료되면 기존 title을 newTitle로 설정하기 위한 것 같은데..
useEffect 없이 issueDetails prop이 변경되면 IssueDetailHeader 렌더링 되도록 하는 방법이 없을까요? 저도 아직 정확한 이유는 정확히 모르겠지만, 공식문서에서 useEffect로 자식의 state를 set하지 말라고 나와있어서 공유드려요.

### 참고
> [prop 변경에 대한 응답으로 특정 state 일부를 조정하려면 렌더링 중에 설정하세요.](https://react-ko.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes)

Kakamotobi replied:
공식문서에서 이 방식이 안좋다고 말한 이유는 prop을 받는 자식 컴포넌트와 그 안에 있는 컴포넌트들이 처음 렌더할 때 stale한 값을 먼저 렌더하고 그 다음에 다시 올바른 값으로 렌더를 하게 되기 때문이라고 하네요.
> This is inefficient because ProfilePage and its children will first render with the stale value, and then render again.

Kakamotobi committed:
#79 fix: issueDetails 업데이트/fetch 완료 시 newTitle 상태 업데이트

- 공식문서의 권고에 따라, `useEffect`을 사용하지 않고 이전 prop을 상태로 두고 과 현재 prop과 비교하여 다르면 `newTitle` 초기 값 업데이트.
